tests/parametric/test_span_links.py

```diff
@@ -42,60 +50,96 @@ def test_span_started_with_link(self, test_agent, test_library):
 
-        assert (link.get("flags") or 0) == 0
+        span_links = json.loads(span.get("meta", {}).get("_dd.span_links"))
+        assert len(span_links) == 1
+        link = span_links[0]
+        root_tid = root.get("meta", {}).get("_dd.p.tid") or "0000000000000000"
+        root_t64 = "{:016x}".format(span["trace_id"])
+        assert link.get("trace_id") == f"{root_tid}{root_t64}"
+        assert link.get("span_id") == "{:016x}".format(root["span_id"])
+        assert link["attributes"].get("foo") == "bar"
+        assert link["attributes"].get("array.0") == "a"
+        assert link["attributes"].get("array.1") == "b"
+        assert link["attributes"].get("array.2") == "c"
+        assert link.get("flags") == 1
 
-    @missing_feature(library="python", reason="test not implemented")
     def test_span_link_from_distributed_datadog_headers(self, test_agent, test_library):
-        """Properly inject datadog distributed tracing information into span links.
+        """Properly inject datadog distributed tracing information into span links when trace_api is v0.4.
         Testing the conversion of x-datadog-* headers to tracestate for
         representation in span links.
         """
         with test_library:
             with test_library.start_span(
                 "root",
-                links=[Link(parent_id=0, attributes={"foo": "bar"})],
-                http_headers=[
-                    ["x-datadog-trace-id", "1234567890"],
-                    ["x-datadog-parent-id", "9876543210"],
-                    ["x-datadog-sampling-priority", "2"],
-                    ["x-datadog-origin", "synthetics"],
-                    ["x-datadog-tags", "_dd.p.dm=-4,_dd.p.tid=0000000000000010"],
+                links=[
+                    Link(
+                        parent_id=0,
+                        attributes={"foo": "bar"},
+                        http_headers=[
+                            ["x-datadog-trace-id", "1234567890"],
+                            ["x-datadog-parent-id", "9876543210"],
+                            ["x-datadog-sampling-priority", "2"],
+                            ["x-datadog-origin", "synthetics"],
+                            ["x-datadog-tags", "_dd.p.dm=-4,_dd.p.tid=0000000000000010"],
+                        ],
+                    )
                 ],
             ):
                 pass
 
-        span = get_span(test_agent)
-        assert span.get("trace_id") != 1234567890
-        assert span_has_no_parent(span)
+        traces = test_agent.wait_for_num_traces(1)
+        span = traces[0][0]
+        assert span_has_no_parent(span) and span.get("trace_id") != 1234567890
         assert span["meta"].get(ORIGIN) is None
-        assert span["meta"].get("_dd.p.dm") == "-4"
-        assert span["metrics"].get(SAMPLING_PRIORITY_KEY) == 2
-
-        assert span.get("span_links") is not None
+        # TODO: create a test case to ensure sampling decisions are propagated by span links
```

Comment:
Feel free to duplicate the test and annotate the copy with @missing_feature then.

Reply:
I moved the assertions to the new test in this commit: f2c3e486c1cab2679372e2ea4f87d5bcb5a6a9e9. 

I also added a test case for span links between two root spans. Does this new test align with the RFC?

With this change python is now passing 6/7 tests. This will allow us to enable more tests and catch regressions in other areas.

Reply:
I also rebased my changes off of @PerfectSlayer's otel system tests. 
Hopefully this PR is now in a mergeable state. 🤞